src/components/computed-properties.vue

```diff
@@ -259,18 +274,21 @@ export default {
       this.emptyForm();
       this.displayList = false;
     },
-    validateDuplicate() {
-      this.existsProperty = false;
-      this.current.forEach(item => {
-        if (item.property === this.add.property && item.id !== this.add.id) {
-          this.existsProperty = true;
+    validateData() {
+      this.ruleDescription = 'required';
+      this.ruleFormula = 'required';
+      this.rulePropName = this.rulePropName ? this.rulePropName : 'required';
+      this.editorInvalid = this.add.formula.trim() === '';
+      let fields = this.$children[0].$children[0].$children;
```

Comment:
This is a little hard to read (what exactly are these deep nested children?) and fragile. If this component's template is refactored, this code could break. I think you could use `$refs` here: https://vuejs.org/v2/api/#vm-refs. Adding the same ref string to multiple components will create an array for you.

Also, you can use `const` here instead of `let` since fields isn't re-assigned.

Reply:
Changed